### PR TITLE
test: move all patch generator unit tests to their own packages

### PR DESCRIPTION
**Pull request by supershal**
This PR is stacked on https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/21
- Move test execution of `region` and `HTTProxy` from Meta patch handler tests. This will help with maintaining unit tests when we add more providers (i.e. Nutanix, GCP, etc..)
- Move test execution in the same unit test file as the test cases. IMHO This will give better control over how individual tests can be run.

### CODE REVIEW

**Review by faiq — APPROVED** [review]

File: test/helpers/environment.go [hunk 1]
```diff
@@ -9,27 +9,12 @@ import (
...
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 var TestEnv *TestEnvironment
 
 // Initialize the test environment. BeforeSuite will be only executed if this package is loaded by the test.
 var _ = BeforeSuite(func(ctx SpecContext) {
-	By("Initialize loggers for testing")
-	// Uninitialized logger spits stacktrace as warning during test execution
```
Comment (supershal) [inline test/helpers/environment.go:23]: I removed this. because this changes are unrelated. I need more testing to add this. I will file separate PR later.

File: common/pkg/testutils/capitest/patches.go [hunk 2]
```diff
@@ -92,6 +92,52 @@ func ValidateGeneratePatches[T mutation.GeneratePatches](
 	DescribeTable("Patches", testFunc, testArgs)
 }
 
+// TODO(shalinpatel): AssertGeneratePatches is a replacement of ValidateGeneratePatches function.
+// remove ValidateGeneratePatches once all the shared test cases are moved to their own patch package.
+func AssertGeneratePatches[T mutation.GeneratePatches](
```
Comment (dkoshkin) [inline common/pkg/testutils/capitest/patches.go:97]: This is similar to ValidateGeneratePatches but does not build a table, why the change? Are there any other differences?

**Review by dkoshkin — APPROVED** [review]

File: pkg/handlers/aws/mutation/region/inject_test.go [hunk 3]
```diff
@@ -25,11 +28,39 @@ var _ = Describe("Generate AWS Region patches", func() {
...
+			ExpectedPatchMatchers: []capitest.JSONPatchMatcher{{
+				Operation:    "add",
+				Path:         "/spec/template/spec/region",
+				ValueMatcher: gomega.Equal("a-specific-region"),
+			}},
+		},
+	}
+
+	// create test node for each case
+	for testIdx := range testDefs {
```
Comment (dkoshkin) [inline pkg/handlers/aws/mutation/region/inject_test.go:56]: So this is now needed because `AssertGeneratePatches` doesn't build the table? This needs to be repeated for every test, why is it being moved out of `ValidateGeneratePatches`?

Comment (supershal) [reply pkg/handlers/aws/mutation/region/inject_test.go:56]: My thinking behind it was to have more control over test execution and keep most of the test execution cycle. for example, If I want to do other additional validation for a httpProxy variable that does not apply to other variables, I can easily add it here.

File: common/pkg/testutils/capitest/patches.go (hunk 2 above)
Comment (supershal) [reply common/pkg/testutils/capitest/patches.go:97]: There is no other changes, it just removes building the table. I plan to remove `ValidateGeneratePatches` or rename `AssertGeneratePatches` to `ValidateGeneratePatches` once all the patches are moved.

**Comment by supershal** [comment]
in any case, if we want to revert these changes, we can revert https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/21 and this PR (https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/23)
